MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -258,7 +258,7 @@ open class RouteProgress: NSObject {
      Returns the SpeedLimit for the current position along the route. Returns SpeedLimit.invalid if the speed limit is unknown or missing.
      */
     public var currentSpeedLimit: SpeedLimit {
-
+        guard legIndex < positionedSpeedLimitsByStep.count, currentLegProgress.stepIndex < positionedSpeedLimitsByStep[legIndex].count else { return .invalid }
```

Comment:
This is a bit late to verify that `positionedSpeedLimitsByStep` has the same number of items as there are legs and that the leg’s array of positioned speed limits has the same number of items as there are steps in the leg. We should assert about those counts after building up `positionedSpeedLimitsByStep`.

Here, all we really need to check is whether `positionedSpeedLimitsByStep` is populated at all. If speed limits can legitimately be missing (as in the case of walking or cycling directions), then `positionedSpeedLimitsByStep` should be optional.